### PR TITLE
docs: add infracost-plan-json example with example test pipeline

### PR DESCRIPTION
**Pull request by hugorut**
Depends on #9

* adds infracost-plan-json example
* adds infracost matrix example
* adds example-test pipeline

Sadly the `template` approach didn't work as templates can't accept variables built at runtime in the pipeline. So there would be no way to run the `generate` script and be able to dynamically create. For the time being I've just added it as a manual step that we need to run - which is exactly how the `actions` repo is setup. However, in the future I think we should add this step in a github-action workflow, then generate the pipeline code and trigger a azure-devops pipeline run from the github-action workflow after successfully building the JSON. This would completely automate the process, but for now I don't think it's necessary, as the examples won't change too often and we can manually generate locally.

### CODE REVIEW

**Comment by hugorut** [comment]
see https://dev.azure.com/infracost/base/_build/results?buildId=327&view=results for example-test pipeline

File: examples/multi-project/README.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Multi-project
+
+These examples show how to run Infracost Azure Devops tasks against a multi-project setup using either an [Infracost config file](https://www.infracost.io/docs/multi_project/config_file) or a GitHub Actions build matrix.
```
Comment (aliscott) [inline examples/multi-project/README.md:3]:
```suggestion
These examples show how to run Infracost Azure Devops tasks against a multi-project setup using either an [Infracost config file](https://www.infracost.io/docs/multi_project/config_file) or the Azure DevOps pipeline matrix strategy.
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (aliscott) [inline package.json:1]: We should keep this consistent with https://github.com/infracost/actions/blob/master/package.json. Especially the LICENSE.

File: scripts/generateExamplesTests.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+ *
+ * @typedef {Object} Pipeline
+ * @property {Job[]} jobs
+ *
+ */
+
+/**
+ * fixupExamples modifies the examples by:
+ * 1. Replacing any infracost/actions steps with the local path
+ * 2. Replacing the infracost/actions/comment step with a step that comment contents
```
Comment (aliscott) [inline scripts/generateExamplesTests.js:62]:
```suggestion
 * fixupExamples modifies the examples by replacing the InfracostComment step with a step that comment contents
```

File: scripts/generateExamplesTests.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+ * @property {string} task
+ * @property {Object} inputs
+ *
+ * @typedef {Object} Job
+ * @property {Step[]} steps
+ *
+ * @typedef {Object} Pipeline
+ * @property {Job[]} jobs
+ *
+ */
```
Comment (aliscott) [inline scripts/generateExamplesTests.js:57]: Nice!

**Review by aliscott — APPROVED** [review]
LGTM!

**Pull request merged**